types/react-hooks-compose/index.d.ts

```diff
@@ -0,0 +1,11 @@
+/// <reference types="react"/>
+
+declare namespace ReactHooksCompose {
```

Comment:
Why the namespace and extra export? Can this just be:

```ts
function composeHooks(): ...

export = composeHooks;
```

Reply:
An excellent question, I went around this a few times as I started with that, but I get the TS error:

```
Top-level declarations in .d.ts files must start with either a 'declare' or 'export' modifier.ts(1046)
```
and exporting the function directly gives the following (it expects the actual function body as it's not considering it a type):

<img width="787" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/394182/56f7452a-fa8e-4bc9-ad8e-33f4d3c9edf5">

So, I ended up finding the namespace declaration worked because it moved from the root level 🤷

Reply:
Ah, you'd just need to add `declare` to what I wrote above (my mistake).

Did you intentionally close this? Happy to continue to review this.

Reply:
Yeah, having battled the type declarations over the weekend some problems went deeper and I decided against pursuing it.

I did learn a lot and greatly appreciate your input... thanks!